apps/nfid-frontend/src/integration/swap/icpswap/impl/shroff-icp-swap-impl.ts

```diff
@@ -79,6 +81,42 @@ export class ShroffIcpSwapImpl extends ShroffAbstract {
     this.swapTransaction = trs
   }
 
+  async getAvailablePools(
```

Comment:
why do you request it on every token? store pairs to memory and request on init method